eval/src/vespa/eval/instruction/generic_concat.cpp

```diff
@@ -93,8 +94,36 @@ void my_generic_concat_op(State &state, uint64_t param_in) {
     state.pop_pop_push(result_ref);
 }
 
+template <typename LCT, typename RCT>
+void my_dense_simple_concat_op(State &state, uint64_t param_in) {
+    using OCT = typename eval::UnifyCellTypes<LCT, RCT>::type;
```

Comment:
if you concat a double and a vector of floats, the result type will be float and not double. This issue seems to also be in the generic version, and not tested in the unit test.